kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ingress/ProxyIngressModel.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator.ingress;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import io.fabric8.kubernetes.api.model.ContainerPort;
+import io.fabric8.kubernetes.api.model.Service;
+import io.fabric8.kubernetes.api.model.ServiceBuilder;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.VirtualKafkaCluster;
+import io.kroxylicious.proxy.config.VirtualClusterGateway;
+
+import static io.kroxylicious.kubernetes.operator.ResourcesUtil.name;
+
+/**
+ * The ProxyIngressModel represents the logical arrangement of ingresses for a single KafkaProxy.
+ * Different ingresses may need different resources on the pod, like unique identifying ports that
+ * the proxy can use to determine the upstream nodes, or a shared port.
+ * This also is a place where we can transform a single KafkaProxyIngress specification into multiple
+ * Ingress instances (numerous virtual clusters can refer to the same ingress), with it own unique
```

Comment:
There are lots of ingresses in this sentence! I'm not sure I understand what the difference between "Ingress" and "ingress", and how they relate to a `ProxyIngressModel`.

Reply:
mm naming is hard, the model class is "an ingress for a particular virtual cluster along with some allocated ports" 

as opposed to KafkaProxyIngress, which you can look at as a sort of template applied to different virtual clusters.